src/generators.cpp

```diff
@@ -301,7 +307,23 @@ void Generator::ComputeLogits() {
   search_->ApplyRepetitionPenalty(search.repetition_penalty);
 }
 
+void Generator::SetRuntimeOptionsConfig(const char* key, const char* value) {
+  if (strcmp(key, "terminate_session") == 0) {
+    if (strcmp(value, "0") == 0 || strcmp(value, "1") == 0) {
+      bool terminate_curr_session = (std::string(value) == "1");
+      // Set value of terminate
+      state_->SetUnsetTerminate(terminate_curr_session);
+    } else {
+      // Value not expected
+      throw std::runtime_error(std::string("terminate_session key value unexpected: ") + value);
+    }
```

Comment:
Use two functions `SetTerminate` and `UnsetTerminate` to make the intention clearer. Alternatively, if you want to use a single function, could you update the name to `SetOrUnsetTerminate`?

```suggestion
    if (strcmp(value, "0") == 0) {
      state_->UnsetTerminate();
    } else if (strcmp(value, "1") == 0) {
      state_->SetTerminate();
    } else {
      // Value not expected
      throw std::runtime_error(std::string("terminate_session key value unexpected: ") + value);
    }
```